homeassistant/helpers/typing.py

```diff
@@ -31,3 +40,12 @@ class UndefinedType(Enum):
 
 
 UNDEFINED = UndefinedType._singleton  # pylint: disable=protected-access
+
+
+class AddEntitiesCallback(Protocol):
```

Comment:
Different opinion.

I don't think we should put this in our `homeassistant.helpers.typing` package. While is feel attractive to do, over time, this package will become bloated.

We didn't put other things in here either? For example, like TypedDicts?

IMHO, this belongs in the `homeassistant.helpers.entity_platform` package (keeping things together).

Reply:
I thought about this. My idea was people more likely will expect to find it here. And they don't usually interact with `homeassistant.helpers.entity_platform`.

But yes, it can be moved there for sure. This will also solve circular dependency issue which I needed to bypass here.

What do you think about name? I find `AddEntitiesCallbackType` too long and IMO `Type` suffix doesn't really provide useful info.

Reply:
Personally, I'm not a fan of using that prefix.

Reply:
Alright, let's move to `homeassistant.helpers.entity_platform` and keep the name.